### PR TITLE
[INLONG-2802][DataProxy] Fix mx.properties local file updating too often

### LINKED ISSUE #2802: [Bug][DataProxy] Update mx.properties local file too often
### What happened

Update mx.properties local file too often

### What you expected to happen

Only when there is a change value, update local file

### How to reproduce

Update mx.properties local file too often

### Environment

_No response_

### InLong version

master

### InLong Component

InLong DataProxy

### Are you willing to submit PR?

- [ ] Yes, I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by pocozh**
### Title Name: [INLONG-2802][DataProxy] Fix mx.properties local file updating too often

Fixes #2802 

### Motivation

*Explain here the context, and why you're making that change. What is the problem you're trying to solve.*

### Modifications

*Describe the modifications you've done.*

### Verifying this change

- [ ] Make sure that the change passes the CI checks.

*(Please pick either of the following options)*

This change is a trivial rework / code cleanup without any test coverage.

*(or)*

This change is already covered by existing tests, such as *(please describe tests)*.

*(or)*

This change added tests and can be verified as follows:

*(example:)*
  - *Added integration tests for end-to-end deployment with large payloads (10MB)*
  - *Extended integration test for recovery after broker failure*

### Documentation

  - Does this pull request introduce a new feature? (yes / no)
  - If yes, how is the feature documented? (not applicable / docs / JavaDocs / not documented)
  - If a feature is not applicable for documentation, explain why?
  - If a feature is not documented yet in this PR, please create a followup issue for adding the documentation

### CODE REVIEW

File: inlong-dataproxy/dataproxy-source/src/main/java/org/apache/inlong/dataproxy/config/ConfigManager.java [hunk 1]
```diff
@@ -123,7 +123,8 @@ private boolean updatePropertiesHolder(Map<String, String> result,
             String oldValue = addElseRemove
                     ? tmpHolder.put(entry.getKey(), entry.getValue()) : tmpHolder.remove(entry.getKey());
             // if addElseRemove is false, that means removing item, changed is true.
-            if (oldValue == null || !oldValue.equals(entry.getValue()) || !addElseRemove) {
+            if ((oldValue == null && entry.getValue() != null) || !oldValue.equals(entry.getValue())
+                    || !addElseRemove) {
```
Comment (luchunliang) [inline inlong-dataproxy/dataproxy-source/src/main/java/org/apache/inlong/dataproxy/config/ConfigManager.java:127]: oldValue.equals(entry.getValue()) maybe throw NullPointerException.
(oldValue == null && entry.getValue() != null) maybe be false when oldValue == null and entry.getValue() == null.

File: inlong-dataproxy/dataproxy-source/src/main/java/org/apache/inlong/dataproxy/config/ConfigManager.java [hunk 2]
```diff
@@ -308,8 +310,12 @@ private boolean checkWithManager(String host, String proxyClusterName) {
                     mqConfig.putAll(clusterSet.get(0).getParams());
```
[the remark this answers is not in the document]
Comment (pocozh) [reply inlong-dataproxy/dataproxy-source/src/main/java/org/apache/inlong/dataproxy/config/ConfigManager.java]: done

**Review by dockerzhang — APPROVED** [review]

File: inlong-dataproxy/dataproxy-source/src/main/java/org/apache/inlong/dataproxy/config/ConfigManager.java (hunk 1 above)
Comment (luchunliang) [reply inlong-dataproxy/dataproxy-source/src/main/java/org/apache/inlong/dataproxy/config/ConfigManager.java:130]: Readability is not good. It is better for the example.
    private boolean updatePropertiesHolder(Map<String, String> result,
            PropertiesConfigHolder holder, boolean addElseRemove) {
        Map<String, String> tmpHolder = holder.forkHolder();
        boolean changed = false;
        for (Map.Entry<String, String> entry : result.entrySet()) {
            if (addElseRemove) {
                String oldValue = tmpHolder.put(entry.getKey(), entry.getValue());
                if (!ObjectUtils.equals(oldValue, entry.getValue())) {
                    changed = true;
                }
            } else {
                String oldValue = tmpHolder.remove(entry.getKey());
                if (oldValue != null) {
                    changed = true;
                }
            }
        }

        if (changed) {
            return holder.loadFromHolderToFile(tmpHolder);
        } else {
            return false;
        }
    }

**Review by luchunliang — APPROVED** [review]

**Review by baomingyu — APPROVED** [review]
+1

**Pull request merged**